File: plugins/title.rs
use std::io;

mod utils;
use utils::text::*;

fn main() {
	let mut lines: Vec<String> = vec![];
	let std_lines = io::stdin().lines();
	for l in std_lines {
		lines.push(l.unwrap());
	}

	for l in &lines {
		let mut title = String::from("Mineral Existence");
		if l.contains("{{title}}") {
			for c in &lines {
				if c.contains("<h2>") {
					title = slice(&c, 4..len(&c) - 5);
					break;
				}
			}
			println!("{}", replace(&l, "{{title}}", &title));
		} else {
			println!("{}", l);
		}
	}
}

File: plugins/title.py
#!/usr/bin/python

import sys
import re

lines = []
match = 0

for line in sys.stdin:
	lines.append(line)

for line in lines:
	if "{{title}}" in line:
		title = "Some Page :)"
		for c_line in lines:
		    #DOC: finds first h2 and uses text inside as title
			if "<h2>" in c_line and match == 0:
				title = re.search(r"<h2>[^<]+<\/h2>", c_line).group()[4:-5]
				match = 1
		line = line.replace("{{title}}", title)
	sys.stdout.write(line)

